src/renderer/rendered_window.rs

```diff
@@ -73,8 +75,8 @@ pub struct RenderedWindow {
     pub grid_size: Dimensions,
 
     has_viewport: bool,
-    scrollback_lines: Vec<Option<Line>>,
-    actual_lines: Vec<Option<Line>>,
+    scrollback_lines: Vec<Option<Arc<Mutex<Line>>>>,
```

Comment:
Would a parkinglot RwLock make more sense here since it looks like its common to read without updating?

Reply:
Leaving for Fredizzimo

Reply:
There's actually no contention at all currently, and the lock is mostly there to satisfy the borrow checker. It can probably be solved some other way, but I think we want to do background shaping at some point, so I left the lock.

But even then, we won't have several concurrent readers, which is the case where you benefit from a RWLock. If there's just one reader and one writer, a normal Mutex simpler and faster.  A RwLock is actually quite complex behind the scenes, see this for example https://eli.thegreenplace.net/2019/implementing-reader-writer-locks/, but with many readers it's faster since nothing need to block.